rotkehlchen/tests/unit/decoders/test_gnosis_sdai.py

```diff
@@ -0,0 +1,245 @@
+            balance=Balance(amount=FVal(sent_amount)),
+            location_label=user_address,
+            notes=f'Redeem {sent_amount} SDAI',
+            counterparty=CPT_SDAI,
+            tx_hash=tx_hash,
+            address=null_address,
+        ),
+    ]
+
+    assert actual_events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('gnosis_accounts', [['0x5938852FE18Ad6963322FB98D1fDDA5c24DD8a0E']])
+def test_deposit_wrapped_sdai(database, gnosis_inquirer, gnosis_accounts):
+    user_address = gnosis_accounts[0]
+    null_address = '0x0000000000000000000000000000000000000000'
+    tx_hash = deserialize_evm_tx_hash('0xd406f40ecd2538d41adb2e645c8fb6d32cec5485510798bfed5d991c258d4b1d')  # noqa: E501
+    actual_events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=gnosis_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1706794335000)
+    withdraw_amount, deposit_amount, gas_amount = '319.006747127200240848', '331.313258668881367296', '0.0006162151'  # noqa: E501
+    expected_events = [
+        EvmEvent(
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_XDAI,
+            balance=Balance(amount=FVal(gas_amount)),
+            location_label=user_address,
+            notes=f'Burned {gas_amount} XDAI for gas',
+            tx_hash=tx_hash,
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            sequence_index=1267,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=A_WXDAI,
+            balance=Balance(amount=FVal(deposit_amount)),
+            location_label=user_address,
+            notes=f'Deposit {deposit_amount} Wrapped XDAI to SDAI contract',
+            tx_hash=tx_hash,
+            counterparty=CPT_SDAI,
+            address='0xD499b51fcFc66bd31248ef4b28d656d67E591A94',
+        ), EvmEvent(
+            sequence_index=1269,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=Asset('eip155:100/erc20:0xaf204776c7245bF4147c2612BF6e5972Ee483701'),
+            balance=Balance(amount=FVal(withdraw_amount)),
+            location_label=user_address,
+            notes=f'Receive {withdraw_amount} SDAI',
+            tx_hash=tx_hash,
+            counterparty=CPT_SDAI,
+            address=null_address,
+        ),
+    ]
+
+    assert actual_events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('gnosis_accounts', [['0x23727b54163F63CffdD8B7769e0eCb13Df253b4e']])
+def test_redeem_wrapped_sdai(database, gnosis_inquirer, gnosis_accounts):
```

Comment:
the name of the test is confusing. You have one that is redeem_sdai and this one is wrapped_dai. Maybe another name would help to understand what this test does.